### ISSUE TITLE
Composer text input not resizing correctly

### ISSUE THREAD

**Issue by pm-dev** [GetStream/stream-chat-swift#3074]
When the text input reaches its maximum height, the text view starts to jitter on each keystroke:
https://github.com/GetStream/stream-chat-swift/assets/3067328/b43946b4-1368-4976-a72d-9d3e89419405

This can be reproduced in the demo app with the following changes:
https://github.com/GetStream/stream-chat-swift/compare/develop...pm-dev:stream-chat-swift:threaded-composer-bug-3

This only happens inside a thread, and requires both the composer's `bottomContainer` to be hidden and the input text view to have it's attributed placeholder set on `updateContent`. 

I can work around this bug by first only setting the attributed placeholder if the text content is empty i.e. `composerView.inputMessageView.textView.text.isEmpty` but this is fragile, because it's unknown if the SDK is guaranteed to call `updateContent` before the placeholder is shown. It would be safer to not use this check and set the placeholder each time.

## GetStream Environment
This seems to be happening across iOS/xcode versions, devices and using the latest version of the SDK, but the specific environment I was using:

**GetStream Chat version:** Latest `develop`
**GetStream Chat frameworks:** StreamChat, StreamChatUI
**iOS version:** 17.2
**Swift version:** 5.9
**Xcode version:** 15.2
**Device:** iPhone 15 simulator

**Comment by nuno-vieira** [comment]
Hi @pm-dev!

Thank you for the report, we will take a look at it as soon as we can 👍 

We will get back to you once we have more details about this one.

Best,
Nuno

**Comment by nuno-vieira** [comment]
Hi @pm-dev,

The issue is related to updating the visibility of the bottom container on the `updateContent()`, we first set it to false, and then you set it true, so this causes a conflict and triggers a UITextView resize really fast. We did encounter this issue already with other customers. It seems like the overriding of `updateContent()`  does not work very well for the composer. 

We will analyse internally what can be done better here, but most likely we will need to find way to only set the values once, and make it overridable by multiple functions instead of everything being done in the `updateContent()` 🤔

**Comment by pm-dev** [comment]
Seems odd that setting the bottom container's visibility causes the UITextView to resize, since the text view is not in the bottom container. Seems like some sort of bug there.

**Comment by nuno-vieira** [comment]
@pm-dev The only explanation that I have right now is that it is caused by this workaround: https://github.com/GetStream/stream-chat-swift/blob/develop/Sources/StreamChatUI/CommonViews/InputTextView/InputTextView.swift#L200-L209

For example, in your code, you are able to fix it if you change it by doing this:
```swift
    override func setUpLayout() {
        super.setUpLayout()

        composerView.inputMessageView.textView.placeholderLabel.attributedText = defaultPlaceholderText
    }

    override func updateContent() {
        super.updateContent()

        composerView.bottomContainer.isHidden = true
    }
```

In your case this should actually be enough, you should not need to update the placeholder in the `updateContent()`

**Comment by pm-dev** [comment]
@nuno-vieira [that's](https://github.com/GetStream/stream-chat-swift/blob/develop/Sources/StreamChatUI/CommonViews/InputTextView/InputTextView.swift#L200-L209) a pretty unfortunate hack/workaround. Unfortunately your suggestion wont work for us because the placeholder is based on the content, so we may need to update it every time content is updated

**Comment by laevandus** [comment]
Hello @pm-dev, we have released a new version [4.51.0](https://github.com/GetStream/stream-chat-swift/releases/tag/4.51.0) which adds a new set of update methods instead of a single `updateContent`.
